orchagent/portsorch.cpp

```diff
@@ -1252,82 +1387,79 @@ bool PortsOrch::removeVlan(Port vlan)
+    sai_status_t status = sai_vlan_api->remove_vlan(vlan.m_vlan_info.vlan_oid);
     if (status != SAI_STATUS_SUCCESS)
     {
-        SWSS_LOG_ERROR("Failed to remove VLAN %s vid:%hu", vlan.m_alias.c_str(), vlan.m_vlan_id);
+        SWSS_LOG_ERROR("Failed to remove VLAN %s vid:%hu",
+                vlan.m_alias.c_str(), vlan.m_vlan_info.vlan_id);
         return false;
     }
 
-    SWSS_LOG_NOTICE("Remove VLAN %s vid:%hu", vlan.m_alias.c_str(), vlan.m_vlan_id);
+    SWSS_LOG_NOTICE("Remove VLAN %s vid:%hu", vlan.m_alias.c_str(),
+		vlan.m_vlan_info.vlan_id);
 
     m_portList.erase(vlan.m_alias);
 
     return true;
 }
 
-bool PortsOrch::addVlanMember(Port vlan, Port port, string& tagging_mode)
+bool PortsOrch::addVlanMember(Port &vlan, Port &port, string &tagging_mode)
 {
     SWSS_LOG_ENTER();
 
     sai_attribute_t attr;
     vector<sai_attribute_t> attrs;
 
     attr.id = SAI_VLAN_MEMBER_ATTR_VLAN_ID;
-    attr.value.oid = vlan.m_vlan_oid;
+    attr.value.oid = vlan.m_vlan_info.vlan_oid;
     attrs.push_back(attr);
 
     attr.id = SAI_VLAN_MEMBER_ATTR_BRIDGE_PORT_ID;
     attr.value.oid = port.m_bridge_port_id;
     attrs.push_back(attr);
 
+    sai_vlan_tagging_mode_t sai_tagging_mode = SAI_VLAN_TAGGING_MODE_TAGGED;
     attr.id = SAI_VLAN_MEMBER_ATTR_VLAN_TAGGING_MODE;
     if (tagging_mode == "untagged")
-        attr.value.s32 = SAI_VLAN_TAGGING_MODE_UNTAGGED;
+        sai_tagging_mode = SAI_VLAN_TAGGING_MODE_UNTAGGED;
     else if (tagging_mode == "tagged")
-        attr.value.s32 = SAI_VLAN_TAGGING_MODE_TAGGED;
+        sai_tagging_mode = SAI_VLAN_TAGGING_MODE_TAGGED;
     else if (tagging_mode == "priority_tagged")
-        attr.value.s32 = SAI_VLAN_TAGGING_MODE_PRIORITY_TAGGED;
+        sai_tagging_mode = SAI_VLAN_TAGGING_MODE_PRIORITY_TAGGED;
     else assert(false);
+    attr.value.s32 = sai_tagging_mode;
     attrs.push_back(attr);
 
     sai_object_id_t vlan_member_id;
     sai_status_t status = sai_vlan_api->create_vlan_member(&vlan_member_id, gSwitchId, (uint32_t)attrs.size(), attrs.data());
     if (status != SAI_STATUS_SUCCESS)
     {
         SWSS_LOG_ERROR("Failed to add member %s to VLAN %s vid:%hu pid:%lx",
-                port.m_alias.c_str(), vlan.m_alias.c_str(), vlan.m_vlan_id, port.m_port_id);
+                port.m_alias.c_str(), vlan.m_alias.c_str(), vlan.m_vlan_info.vlan_id, port.m_port_id);
         return false;
     }
     SWSS_LOG_NOTICE("Add member %s to VLAN %s vid:%hu pid%lx",
-            port.m_alias.c_str(), vlan.m_alias.c_str(), vlan.m_vlan_id, port.m_port_id);
+            port.m_alias.c_str(), vlan.m_alias.c_str(), vlan.m_vlan_info.vlan_id, port.m_port_id);
 
-    if (tagging_mode == "untagged") // set pvlan id for untagged port only
+    /* Use untagged VLAN as pvid of the member port when no pvid has been configured explictly */
+    if (port.m_port_vlan_id == DEFAULT_PORT_VLAN_ID && sai_tagging_mode == SAI_VLAN_TAGGING_MODE_UNTAGGED)
```

Comment:
>port.m_port_vlan_id == DEFAULT_PORT_VLAN_ID  [](start = 8, length = 44)

not sure why we need this condition, can you explain?

Reply:
As explained previously, it was for supporting pvid configuration from admin interface.  
removed.